### PR TITLE
Fix: JavaScript Fix error when searching after load() and view()

### PR DESCRIPTION
**Pull request by abetomo**
I got an error when I loaded and searched with load() or view().

Code Example:
```js
// Saved with `index.save('index.usearch');` in another script.
index.load('index.usearch');
const results = index.search(new Float32Array([0.2, 0.6, 0.4]), 10);
```

Error:
```
/build/javascript/dist/cjs/usearch.js:314
        const result = __classPrivateFieldGet(this, _Index_compiledIndex, "f").search(normalizedVectors, k);
                                                                               ^

TypeError: Search failed
    at Index.search (/build/javascript/dist/cjs/usearch.js:314:80)
    at Object.<anonymous> (/build/example.js:9:21)
    at Module._compile (node:internal/modules/cjs/loader:1469:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1548:10)
    at Module.load (node:internal/modules/cjs/loader:1288:32)
    at Module._load (node:internal/modules/cjs/loader:1104:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:174:12)
    at node:internal/main/run_main_module:28:49

Node.js v20.17.0
```

This bug fixed.

### CODE REVIEW

**Comment by DavidGOrtega** [comment]
@abetomo great!! Im suffering the same.
Can someone please  check this?

**Comment by ashvardanian** [comment]
There is a compilation issue on Clang, that needs to be resolved to merge this, @DavidGOrtega & @abetomo. Will do my best to merge tomorrow. Thanks for contributions, guys!

File: javascript/lib.cpp [hunk 1]
```diff
@@ -129,6 +129,7 @@ void CompiledIndex::Load(Napi::CallbackInfo const& ctx) {
         auto result = native_->load(path.c_str());
         if (!result)
             Napi::TypeError::New(ctx.Env(), result.error.release()).ThrowAsJavaScriptException();
+        native_->reserve(ceil2(native_->size()));
```
Comment (ashvardanian) [inline javascript/lib.cpp:132]: I believe in this context it's cleaner to use `try_reserve`, that return a boolean, and if it's `false` - propagate an `Napi` error.

**Review by ashvardanian — CHANGES_REQUESTED** [review]

**Comment by ashvardanian** [comment]
@abetomo, thank you for being so active, responsive, and ready to operate across languages! It’s great to have you on our side 🤗

**Comment by abetomo** [comment]
Thanks for your review!
I would like to continue to contribute.

**Pull request merged**